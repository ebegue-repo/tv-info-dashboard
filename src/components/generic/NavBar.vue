<template>
  <div class="min-h-full">
    <nav class="bg-tvm-green">
      <div class="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">
        <div class="flex h-16 items-center justify-between">
          <div class="flex items-center">
            <div class="flex-shrink-0">
              <img class="h-8 w-8" src="../../assets/images/tv_logo.png">
            </div>
            <div>
              <div class="ml-10 flex items-baseline space-x-6">
                <router-link class="text-white" v-for="item in navigation" :to="{ name: item.name }"
                             :key="item.name">
                  {{ item.name }}
                </router-link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </div>
</template>


<script setup lang="ts">

  defineProps({
    navigation: {
      type: Array,
      required: true
    }
  });

</script>
