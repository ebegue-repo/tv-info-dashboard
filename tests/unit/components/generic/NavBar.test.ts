import { mount } from '@vue/test-utils';
import NavBar from '@/components/generic/NavBar.vue';
import { describe, expect, test, vi } from 'vitest';

describe('NavBar', () => {
    test('It should render the generic navigation bar with title "test label" and match the snapshot', () => {

        const wrapper = mount(NavBar, {
            props: {
                navigation: ["test1", "test2", "test3"]
            }
        });
        
        expect(wrapper.html()).toMatchSnapshot();
    });
});