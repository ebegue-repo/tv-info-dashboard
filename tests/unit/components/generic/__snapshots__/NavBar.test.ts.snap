// Vitest Snapshot v1

exports[`NavBar > It should render the generic navigation bar with title "test label" and match the snapshot 1`] = `
"<div class=\\"min-h-full\\">
  <nav class=\\"bg-tvm-green\\">
    <div class=\\"mx-auto max-w-7xl px-4 sm:px-6 lg:px-8\\">
      <div class=\\"flex h-16 items-center justify-between\\">
        <div class=\\"flex items-center\\">
          <div class=\\"flex-shrink-0\\"><img class=\\"h-8 w-8\\" src=\\"/src/assets/images/tv_logo.png\\"></div>
          <div>
            <div class=\\"ml-10 flex items-baseline space-x-6\\">
              <router-link class=\\"text-white\\" to=\\"[object Object]\\"></router-link>
              <router-link class=\\"text-white\\" to=\\"[object Object]\\"></router-link>
              <router-link class=\\"text-white\\" to=\\"[object Object]\\"></router-link>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>"
`;
